modules/likes/jetpack-likes-settings.php

```diff
@@ -590,22 +590,22 @@ function is_comments_enabled() {
 	function admin_settings_callback() {
 		// We're looking for these, and doing a dance to set some stats and save
 		// them together in array option.
-		$new_state = ! empty( $_POST['wpl_default'] ) ? $_POST['wpl_default'] : 'on';
+		$new_state = ! empty( $_POST['wpl_default'] ) ? $_POST['wpl_default'] : false;
 		$db_state  = $this->is_enabled_sitewide();
 
-		$reblogs_new_state = ! empty( $_POST['jetpack_reblogs_enabled'] ) ? $_POST['jetpack_reblogs_enabled'] : 'on';
+		$reblogs_new_state = ! empty( $_POST['jetpack_reblogs_enabled'] ) ? $_POST['jetpack_reblogs_enabled'] : false;
 		$reblogs_db_state = $this->reblogs_enabled_sitewide();
 		/** Default State *********************************************************/
 
 		// Checked (enabled)
-		switch( $new_state ) {
-			case 'off' :
+		switch ( (bool) $new_state ) {
```

Comment:
This will not work the way you expect it to. You can try it yourself with a snippet like this:
```
<?php
echo (bool) 'off';
```
You'll see that a string `off` is typecast as `true` instead of `false`.

Reply:
Hm...since I changed the values to `0` and `1` in the inputs (lines 497 / 503 and 517 / 523), wouldn't casting those as boolean result in `false` / `true`?